src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritConnection.java

```diff
@@ -94,6 +94,7 @@
     private AuthenticationUpdater authenticationUpdater = null;
     private final Set<ConnectionListener> listeners = new CopyOnWriteArraySet<ConnectionListener>();
     private int sshRxBufferSize = SSH_RX_BUFFER_SIZE;
+    private StringBuilder eventBuffer = null;
```

Comment:
> It is an instance variable and the only place where it is created, modified or deleted is in the private getLine method.

Agree if such cycle is completed within one method call. But created instance is stored into a GerritConnection instance field then return a method call. Unfortunately it is not guaranteed that getLine method call is done only by the same thread during keeping this StringBuilder instance. My sample code might reproduce such situation.